src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/topology/AbstractTopology.java

```diff
@@ -285,13 +286,22 @@ protected SpoutDeclarer declareKafkaSpout(TopologyBuilder builder, List<String>
     protected SpoutDeclarer declareSpout(TopologyBuilder builder, IRichSpout spout, String spoutId) {
         Integer spoutParallelism = null;
         Integer spoutNumTasks = null;
+        Integer spoutMaxSpoutPending = null;
         if (topologyDef != null) {
             SpoutDef spoutDef = topologyDef.getSpoutDef(spoutId);
             if (spoutDef != null) {
                 spoutParallelism = spoutDef.getParallelism();
                 if (spoutDef.getNumTasks() > 0) {
                     spoutNumTasks = spoutDef.getNumTasks();
                 }
+                if (spoutDef.getProperties() != null) {
+                    for (PropertyDef propertyDef : spoutDef.getProperties()) {
+                        if (propertyDef.getName().equals("max.spout.pending")) {
```

Comment:
maybe `max.spout.pending` should be constant?
what if property is unset? should we use
`"max.spout.pending".equals(propertyDef.getName())` to handle NPE?

Reply:
It could be a constant. Changed.